keras_cv/utils/train.py

```diff
@@ -51,3 +51,29 @@ def convert_inputs_to_tf_dataset(
     if batch_size is not None:
         dataset = dataset.batch(batch_size)
     return dataset
+
+
+def get_feature_extractor(model, layer_names, output_keys=None):
```

Comment:
I really like how concise this function is.  

Nit: Maybe this utility belongs in `backbones` instead of `utils/train`? (if you dont intend this to be user facing feel free to ignore this comment - we are moving to a generated API soon anyways)

Reply:
I do intend this to be user facing. We can always change the imports in `__init__.py`. 

The reason I removed from the `Backbone` class is
1. This is completely general and could be used in other contexts
2. A custom backbone currently only needs to implement `pyramid_level_inputs`. Having to attach this method as well seems silly.